shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.h

```diff
@@ -26,11 +34,65 @@
  *
  * The notify block is guaranteed to be called within a core animation transaction.
  */
-- (void)performCommit:(CGSize)size notify:(nonnull dispatch_block_t)notify;
+- (void)performCommitForView:(int64_t)viewId
+                        size:(CGSize)size
+                      notify:(nonnull dispatch_block_t)notify;
+
+/**
+ * Requests the synchronizer to track another view.
+ *
+ * A view must be registered before calling begineResizeForView: or
+ * performCommitForView:. It is typically done when the view controller is
+ * created.
+ */
+- (void)registerView:(int64_t)viewId;
+
+/**
+ * Requests the synchronizer to no longer track a view.
+ *
+ * It is typically done when the view controller is created.
```

Comment:
The comment seems wrong. Also deregisterView does not seem to be called anywhere. Should it be called from `[FlutterView dealloc]`?

Reply:
I fixed the comment, and removed register & deregister to `FVC` which handles lifecycle better.